lib/logpipe.h

```diff
@@ -214,16 +214,31 @@ struct _LogPipe
 {
   GAtomicCounter ref_cnt;
   gint32 flags;
+
+  /* moving queue to the front should improve performance somewhat, although
+   * I didn't measure it at all.  Rationale: the pointer pointing to this
+   * LogPipe instance is resolved right before calling queue and the
+   * colocation of flags and the queue pointer ensures that they are on the
+   * same cache line, and since the CPU returns memory reads sequentially,
+   * it is among the first values that are read ahead.  It might need to be
+   * measured sometime, but this method is probably the single most often
+   * called virual method, so I felt, that even though the optimization is
+   * probably premature without explicit tests, I can do this simply by
```

Comment:
I tend to agree this might reduce cache use, because the skipped fields are five pointers = 40bytes is relately high compared to an i7 cache line of 64byte, so the chance they end up on different cache lines are not negliglible.

With similar reasoning, do you think it would worth move `pipe_next` upwards as well? log_pipe_forward is called at the same time as ->queue just on the other branch on the if.

At the same time, I doubt we could measure the difference: we are talking about cache sizes of 256kb of L1 and 4mb of L2, so using one extra cache line should not mean too much difference. But to be safe, I do not problem with the rearrangement, it can only improve the performance.

However, this comment is in a **very** bad position in the code. It looks as if it would be a `docstring` explaining how LogPipe works, but it is about a very low level optimization, that should not grab the focus for typical investigations. I was just wandering here with gqueue push pop context in my head, and I could only realize this is about rearranging fields only after I found the patch with git blame. We should delete this comment as soon as possible. It would have been a good comment in the commit message though...